ProjectBank/Server.Tests/Controllers/ProjectsControllerTests.cs

```diff
@@ -0,0 +1,133 @@
+namespace Server.Tests;
+
+public class ProjectsControllerTests
+{
+
+    [Fact]
+    public async Task Post_creates_Project()
+    {
+        // Arrange
+        var toCreate = new CreateProjectDTO();
+        var created = new ProjectDTO(1, "Very nice project", "Active", "1", "A created project", DateTime.UtcNow, DateTime.UtcNow, new List<string>(), new List<UserDTO>());
+        var repository = new Mock<IProjectRepository>();
+        repository.Setup(m => m.CreateProjectAsync(toCreate)).ReturnsAsync(created);
+        var controller = new ProjectsController(repository.Object) { GetObjectId = () => "1" };
+
+        // Act
+        var result = await controller.Post(toCreate) as CreatedAtActionResult;
+
+        // Assert
+        Assert.Equal(created, result?.Value);
+        Assert.Equal("Get", result?.ActionName);
+        Assert.Equal(KeyValuePair.Create("Id", (object?)1), result?.RouteValues?.Single());
+    }
+
+    [Fact]
+    public async Task Get_returns_Projects_from_repo()
+    {
+        // Arrange
+        var expected = Array.Empty<ProjectDTO>();
+        var repository = new Mock<IProjectRepository>();
+        repository.Setup(m => m.ReadAllAsync()).ReturnsAsync(expected);
+        var controller = new ProjectsController(repository.Object);
+
+        // Act
+        var actual = await controller.Get();
+
+        // Assert
+        Assert.Equal(expected, actual);

```

Comment:
Here we also expect zero projects?

Reply:
Same arguments as before. Rasmus also does this is his tests